src/full/Agda/TypeChecking/Level.hs

```diff
@@ -124,7 +124,7 @@ unlevelWithKit :: LevelKit -> Level -> Term
 unlevelWithKit LevelKit{ lvlZero = zer, lvlSuc = suc, lvlMax = max } = \case
   Max m []  -> unConstV zer suc m
   Max 0 [a] -> unPlusV suc a
-  Max m as  -> foldl1 max $ [ unConstV zer suc m | m > 0 ] ++ map (unPlusV suc) as
+  Max m as  -> maximum $ [ unConstV zer suc m | m > 0 ] ++ map (unPlusV suc) as
```

Comment:
`max` is locally bound here, is not the Ord one!

Reply:
Opted to use the record pun instead, `lvlMax`. This doesn't trigger the HLint suggestion.